### ISSUE TITLE
Do not warn of setter-related exceptions when getting a property value

### ISSUE THREAD

**Issue by cmeeren** [CSharpAnalyzers/ExceptionalReSharper#49]
```c#
foreach (object key in Data.Keys)
{
    object value = Data[key];  // Data is an IDictionary
    // ...
}
```
The code above warns of `NotSupportedException`, which occurs when setting a value, not getting it. (Specifically, in this case, the exception is thrown when setting a value on a read-only dict, or when the object has a fixed size and the key doesn't exist.)

**Comment by kirbatious** [comment]
I agree, this is the main reason why I use // ReSharper Disable Once ExceptionNotDocumented. Would be lovely to get a fix for this at some point.
